tests/app/main/test_frameworks.py

```diff
@@ -5758,3 +5757,209 @@ def test_false_values(self, falsey_value):
+        framework_slug = 'g-cloud-12'
+        self.login()
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug=framework_slug,
+                                                                         framework_agreement_version="1")
+        self.data_api_client.find_draft_services_iter.return_value = [
+            {'serviceName': 'A service', 'status': 'submitted', 'lotSlug': 'iaas'}
+        ]
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=True)
+
+        res = self.client.get(f"/suppliers/frameworks/{framework_slug}/start-framework-agreement-signing")
+        assert res.status_code == 200
+        res = self.client.post(f"/suppliers/frameworks/{framework_slug}/start-framework-agreement-signing",
+                               data={'legal_authority': 'no'})
+        assert res.status_code == 200
+        assert "You cannot sign the Framework Agreement" in res.get_data(as_text=True)
+
+    def test_post_no_response_shows_error(self):
+        framework_slug = 'g-cloud-12'
+        self.login()
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug=framework_slug,
+                                                                         framework_agreement_version="1")
+        self.data_api_client.find_draft_services_iter.return_value = [
+            {'serviceName': 'A service', 'status': 'submitted', 'lotSlug': 'iaas'}
+        ]
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=True)
+        res = self.client.post(f"/suppliers/frameworks/{framework_slug}/start-framework-agreement-signing",
+                               data={})
+        assert res.status_code == 400
+        assert "Select yes if you have the legal authority to sign on behalf of your company" in res.get_data(
+            as_text=True)
+
+
+class TestSignFrameworkAgreement(BaseApplicationTest):
+    """Tests for app.main.views.frameworks.sign_framework_agreement"""
+
+    def setup_method(self, method):
+        super().setup_method(method)
+        self.data_api_client_patch = mock.patch('app.main.views.frameworks.data_api_client', autospec=True)
+        self.data_api_client = self.data_api_client_patch.start()
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super().teardown_method(method)
+
+    @pytest.mark.parametrize(
+        ('framework_slug', 'on_framework', 'status_code'),
+        (
+            ('g-cloud-11', True, 404),
+            ('g-cloud-12', True, 200),
+            ('g-cloud-12', False, 400),
+        )
+    )
+    def test_only_works_for_supported_frameworks(self, framework_slug, on_framework, status_code):
```

Comment:
Could we also test the page works for both `live` and `standstill` framework statuses? Probably better off in a separate test, something like:

```
@pytest.mark.parametrize(
        ('framework_status', 'status_code'),
        (
            ('coming', 404),
            ('open', 404),
            ('pending', 404),
            ('standstill', 200),
            ('live', 200),
            ('expired', 404),
        )
    )
    def test_only_works_for_live_and_standstill_frameworks(self, framework_slug, status_code):
        ...
``` 

Could do with having a test like this for the `/start-framework-agreement-signing` view too?

Reply:
I am not sure we enforce this currently but in adding a test we can confirm and enforce if necessary.

Reply:
fixed in 7c1f9f2